Consul/Session.cs

```diff
@@ -101,7 +101,7 @@ public class SessionEntry
         public List<string> Checks { get; set; }
 
         [JsonConverter(typeof(NanoSecTimespanConverter))]
-        public TimeSpan LockDelay { get; set; }
+        public TimeSpan? LockDelay { get; set; }
```

Comment:
Since you're making this a nullable to signal "no opinion on how long lockdelay should be", would `[JsonProperty(NullValueHandling = NullValueHandling.Ignore)]` work here instead checking the HasValue property below? This means we could drop the ShouldSerializeLockDelay method entirely.